File: lab7/log_probability.pl
#!/usr/bin/perl -w
 $pattern = "$ARGV[0]";
 @files = glob("lyrics/*.txt");

foreach  $file (@files){
	$matches = 0;
	$wordcount = 0;
	open (F,'<', $file) or die;
	while($line = <F>){
	
	$wordcount += (@words) = $line =~ m/[a-zA-Z]+/g;
	foreach  $word (@words){
			if ($word =~ m/^$pattern$/i){
				$matches++;
			}
		}
	}
	
	$file =~ s/.txt//;
	$file =~ s/lyrics\///;
	$file =~ s/_/ /g;
	
	$matches++;
	$log = log($matches/$wordcount);
	printf "log((%d+1)/%6d) = %8.4f %s\n",$matches-1,$wordcount,$log,$file;
	close F;
}

File: lab7/log_probability.py
#!/usr/bin/python 
import sys
import re
import math
import glob


pattern = sys.argv[1].lower()

lyrics = glob.glob('lyrics/*.txt')
lyrics.sort()
for lyric in lyrics:
    wordCount = 0
    matches = 0
    F = open(lyric,"r")
    for line in F:
        words = re.findall('[a-zA-Z]+',line,flags=0)
        for word in words:
            if word.lower() == pattern:
                matches +=1
        wordCount += len(words);
    artist = re.sub('.txt','',lyric,count=0,flags=0)
    artist = re.sub('lyrics\/','',artist,count=0,flags=0)
    artist = re.sub("_"," ",artist,count=0,flags=0)
    
    frequency = float(matches+1)/float(wordCount)
    logProb = math.log(frequency) 
    print "log(("+str(matches)+"+1)/ "+str(wordCount)+") = ",'{:.4f}'.format(logProb),artist
    F.close()

